opm/simulators/wells/WellBhpThpCalculator.cpp

```diff
@@ -839,6 +866,114 @@ bruteForceBracket(const std::function<double(const double)>& eq,
+
+    const double aqua = rates[BlackoilPhases::Aqua];
+    const double liquid = rates[BlackoilPhases::Liquid];
+    const double vapour = rates[BlackoilPhases::Vapour];
+    double flo = detail::getFlo(table, aqua, liquid, vapour);
+    double wfr, gfr;
+    if (well_.useVfpExplicit() || -flo < table.getFloAxis().front()) {
+        wfr =  well_.vfpProperties()->getExplicitWFR(controls.vfp_table_number, well_.indexOfWell());
+        gfr = well_.vfpProperties()->getExplicitGFR(controls.vfp_table_number, well_.indexOfWell());
+    } else {
+        wfr = detail::getWFR(table, aqua, liquid, vapour);  
+        gfr = detail::getGFR(table, aqua, liquid, vapour);   
+    }
+    if (wfr <= 0.0 && gfr <= 0.0) {
+        // warning message
+    }
+    auto ipr = getFloIPR(well_state, well, summaryState);
+    
+    const double vfp_ref_depth = well_.vfpProperties()->getProd()->getTable(controls.vfp_table_number).getDatumDepth();
+
+    const double dp_hydro = wellhelpers::computeHydrostaticCorrection(well_.refDepth(), vfp_ref_depth, 
+                                                                      rho, well_.gravity());
+    if (ipr.first <= 0.0) {
+        // error message
+    }
+    auto bhp_adjusted = [this, &thp, &dp_hydro](const double bhp) {
+           return bhp - dp_hydro + getVfpBhpAdjustment(bhp, thp);
+       };
+    const auto retval = detail::intersectWithIPR(table, thp, wfr, gfr, well_.getALQ(well_state), ipr.first, ipr.second, bhp_adjusted);
+    if (retval.has_value()) {
+        // returned pair is (flo, bhp)
+        return retval.value().second;
+    } else {
+        return std::nullopt;
+    }
+}    
+
+std::pair<double, double> WellBhpThpCalculator::
+getFloIPR(const WellState& well_state,
+          const Well& well, 
+          const SummaryState& summary_state) const 
+{
+    std::pair<double,double>retval(0.0, 0.0);
+    const auto& controls = well.productionControls(summary_state);
+    const auto& table = well_.vfpProperties()->getProd()->getTable(controls.vfp_table_number);
+    const auto& pu = well_.phaseUsage();
+    const auto& ipr_a= well_state.well(well_.indexOfWell()).implicit_ipr_a;
+    const auto& aqua_a = pu.phase_used[BlackoilPhases::Aqua]? ipr_a[pu.phase_pos[BlackoilPhases::Aqua]]:0.0;
+    const auto& liquid_a = pu.phase_used[BlackoilPhases::Liquid]? ipr_a[pu.phase_pos[BlackoilPhases::Liquid]]:0.0;
+    const auto& vapour_a = pu.phase_used[BlackoilPhases::Vapour]? ipr_a[pu.phase_pos[BlackoilPhases::Vapour]]:0.0;
+    const auto& ipr_b= well_state.well(well_.indexOfWell()).implicit_ipr_b;
+    const auto& aqua_b = pu.phase_used[BlackoilPhases::Aqua]? ipr_b[pu.phase_pos[BlackoilPhases::Aqua]]:0.0;
+    const auto& liquid_b = pu.phase_used[BlackoilPhases::Liquid]? ipr_b[pu.phase_pos[BlackoilPhases::Liquid]]:0.0;
+    const auto& vapour_b = pu.phase_used[BlackoilPhases::Vapour]? ipr_b[pu.phase_pos[BlackoilPhases::Vapour]]:0.0;
+    return std::make_pair(detail::getFlo(table, aqua_a, liquid_a, vapour_a), 
```

Comment:
I had to think a while before I understood the point here: to use the getFlo() helper to take into account the type of the flo variable (oil, liquid, or gas) when making the a and b variables. Suggest adding doc to the function, and also this use of getFlo().

Reply:
Agree, added some doc on this